client/hooks/useTestPlanRunValidatedAssertionCounts.js

```diff
@@ -0,0 +1,99 @@
+import { useMemo } from 'react';
+import { isAssertionValidated } from '../utils/automation';
+import { useQuery } from '@apollo/client';
+import { TEST_PLAN_RUN_ASSERTION_RESULTS_QUERY } from '../components/TestQueueCompletionStatusListItem/queries';
+
+export const useTestPlanRunValidatedAssertionCounts = (
+    testPlanRun,
+    pollInterval = null
+) => {
+    const {
+        data: testPlanRunAssertionsQueryResult,
+        startPolling,
+        stopPolling,
+        refetch
+    } = useQuery(TEST_PLAN_RUN_ASSERTION_RESULTS_QUERY, {
+        variables: {
+            testPlanRunId: testPlanRun.id
+        },
+        fetchPolicy: 'cache-and-network',
+        pollInterval
+    });
+
+    const testResultsLength = useMemo(() => {
+        if (testPlanRunAssertionsQueryResult) {
+            const length =
+                testPlanRunAssertionsQueryResult.testPlanRun?.testResults
+                    ?.length;
+            return length;
+        } else {
+            return 0;
+        }
+    }, [testPlanRunAssertionsQueryResult, stopPolling, startPolling]);
```

Comment:
This code returns `undefined` if an optional property is undefined. If the properties are always present, then we should remove the optional chaining. Otherwise, it seems like we'd want the zero fallback:

```suggestion
    const testResultsLength = useMemo(() => {
        return testPlanRunAssertionsQueryResult?.testPlanRun?.testResults?.length || 0;
    }, [testPlanRunAssertionsQueryResult, stopPolling, startPolling]);
```

Separately, it doesn't seem like `stopPolling` and `startPolling` are dependencies for this calculation. Is it safe to remove them?